roles/StackStorm.ewc/tasks/main.yml

```diff
@@ -24,6 +24,8 @@
     - ewc
     - st2 enterprise
     - skip_ansible_lint
+  notify:
+    - restart st2api
```

Comment:
Because `bwc-enterprise` package also includes LDAP auth backend, are you sure `st2api` restart is sufficient or `st2auth restart` is needed too?

Reply:
Good catch - we should also restart st2auth.

This should happen after configuring auth backed in st2.cond though

Reply:
That already happens https://github.com/StackStorm/ansible-st2/blob/master/roles/StackStorm.ewc/tasks/ldap.yml, however edge case is when the installation wasn't managed by Ansible before or user changed st2.conf by hand/uninstalled/installed/upgraded enterprise package.

Reply:
https://github.com/StackStorm/ansible-st2/blob/ae27046e29ed9f890a3b138512fbd3f66dcb30b2/roles/StackStorm.ewc/tasks/ldap.yml#L13-L16

`st2stream` restart might be needed, too?

Reply:
nope, it should only affect st2auth. st2api and st2stream only work with auth token / API key and are not aware of any auth backed related stuff.

Reply:
And after more thought and looking at the code - I don't think this need or should happen here.

LDAP backend is not used and enabled by default when installing ``bwc-enterprise`` (aka that's different than ``st2-rbac-backend`` pacakge).

So st2api and st2auth needing to be restarted is purely depended on the ldap playbook - when ldap is configured and configuration changes.

Reply:
See previous comment: 
> however edge case is when the installation wasn't managed by Ansible before or user changed st2.conf by hand/uninstalled/installed/upgraded enterprise package

and so `st2auth` restart should happen here as well.

This will cover case when user already had settings for LDAP configured in `st2.conf` (which doesn't trigger st2auth restart), he then upgraded the enterprise package to new version. In such scenario old py code will be used instead of new code of ldap package.